### PR TITLE
Proper collisions while manipulating

### PR DESCRIPTION
**Pull request by MenelvagorMilsom**
## Overview
Previously while manipulating a rigidbody object there was no collisions with static colliders, and the collisions with rigidbody objects were jittery and incorrect. Also, making a bounding box object into a rigidbody would in some cases cause the bounding box to collide with its own rotate/scale handles.

These changes ensure that rigidbody manipulation gives the manipulated object proper velocities and angular velocities, so that unity can do proper collision response. The velocities are assigned to emulate existing smoothing.

## Changes
- `ObjectManipulator` uses lerp/slerp when no rigidbody attached, else uses velocity/angular velocity.
- Ensure that `BoundingBox` rigRoot is has the target object as its parent.
- Add new tests for new collision behaviour.
- Add sample scene.
- Fixes: #6268.

### Before
no_collisions

### After
yes_collisions

### CODE REVIEW

File: Assets/MixedRealityToolkit.Examples/Experimental/Demos/UX/ObjectManipulator/Scripts/ReturnToBounds.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+﻿using System.Collections;
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Examples/Experimental/Demos/UX/ObjectManipulator/Scripts/ReturnToBounds.cs]: Missing [license header](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#script-license-information-headers)

File: Assets/MixedRealityToolkit.Examples/Experimental/Demos/UX/ObjectManipulator/Scripts/ReturnToBounds.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ReturnToBounds : MonoBehaviour
+{
+    [SerializeField]
+    Transform frontBounds = null;
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Examples/Experimental/Demos/UX/ObjectManipulator/Scripts/ReturnToBounds.cs]: Missing [access modifiers](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#access-modifiers)

File: Assets/MixedRealityToolkit.SDK/Experimental/Features/Input/Handlers/ObjectManipulator.cs [hunk 3]
```diff
@@ -686,10 +689,10 @@ private void HandleManipulationStarted()
             if (rigidBody != null)
             {
                 wasKinematic = rigidBody.isKinematic;
-                rigidBody.isKinematic = true;
+                rigidBody.isKinematic = false;
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.SDK/Experimental/Features/Input/Handlers/ObjectManipulator.cs]: Do we want to make this new behavior optional? Or is the presence of the rigidbody now the "opt-in" for this feature?

Comment (keveleigh) [reply Assets/MixedRealityToolkit.SDK/Experimental/Features/Input/Handlers/ObjectManipulator.cs]: That works for me!

File: Assets/MixedRealityToolkit.Examples/Experimental/Demos/UX/ObjectManipulator/Scripts/ReturnToBounds.cs (hunk 2 above)
Comment (MenelvagorMilsom) [reply Assets/MixedRealityToolkit.Examples/Experimental/Demos/UX/ObjectManipulator/Scripts/ReturnToBounds.cs]: Is this required even for demo scripts?

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Examples/Experimental/Demos/UX/ObjectManipulator/Scripts/ReturnToBounds.cs]: I've been trying to treat all code that goes in the repo with the same quality, regardless of the module it ends up in

Comment (MenelvagorMilsom) [reply Assets/MixedRealityToolkit.Examples/Experimental/Demos/UX/ObjectManipulator/Scripts/ReturnToBounds.cs]: Cool, sounds good to me

**Comment by MenelvagorMilsom** [comment]
Looks like theres a failing Bounding Box test. This is into `feature/ux_refactor` so it's not pressing to get it in for 2.2. I'll have a look on Monday.

**Pull request merged**